### PR TITLE
Adds tests for environment.toWebpackConfig

### PR DESCRIPTION
**Pull request by rossta**
As a follow up to #1097 and #1098, I've taken a stab at adding tests for `environment.toWebpackConfig`. 

Because the `environment` module assumes the location of `config/webpacker.yml`, adding tests for this module presents a challenge since this location does not exists relative to the root of the project. Possible approaches include simply adding `config/webpacker.yml` to the root of the project (and potentially other app files, like `app/javascript/packs`), generating files during the test run, or using a mock filesystem. 

The approach I landed on makes it possible to customize the `RAILS_ROOT` location so that the environment paths resolve relative to `test/test_app` in tests. While approach adds flexibility, it also means ensuring that relative paths, e.g. `node_modules`, resolve relative to the given root if it exists.

Let me know what you think of this approach.

### CODE REVIEW

File: package/root.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { resolve, join } = require('path')
+const root = process.env.RAILS_ROOT || '';
```
Comment (gauravtiwari) [inline package/root.js]: What do you think about using `process.cwd()`?

File: package/environment.js [hunk 2]
```diff
@@ -77,7 +78,7 @@ const getBaseConfig = () =>
     },
 
     resolveLoader: {
-      modules: ['node_modules']
+      modules: [join(root, 'node_modules')]
```
Comment (gauravtiwari) [inline package/environment.js]: Don't think this is necessary??

**Comment by rossta** [comment]
Turns out we can change the working directory of the test process, taking advantage of how we execute requires of the default and project webpacker.yml files, without changing application code. 

I'm not sure how useful the original approach of introducing a configurable Rails root would be. In any case, it should probably be a separate PR rather than a side effect of trying to test existing app code.

File: package/utils/__tests__/objectify.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 /* global test expect */
+console.log('objectify test cwd', process.cwd());
```
Comment (gauravtiwari) [inline package/utils/__tests__/objectify.js]: Is this intentional?

**Comment by gauravtiwari** [comment]
Awesome @rossta 💯  🍰  

Just one minor comment and then we can merge. Thank you :)

File: package/utils/__tests__/objectify.js (hunk 3 above)
Comment (rossta) [reply package/utils/__tests__/objectify.js]: No, that was a debugging line that should be removed (and looks like you've done that on master already).

**Comment by rossta** [comment]
@gauravtiwari 👍

**Pull request merged**